### ISSUE TITLE
[toc2] Why is toc2 limited to input areas?

### ISSUE THREAD

**Issue by mpschr** [ipython-contrib/jupyter_contrib_nbextensions#1339]
I quite often produce various outputs from a line of code which I structure with `<h>` elements for better navigation as follows:

```
from IPython.core.display import HTML, display
for x in somelist:
    t = *some code*
    display(HTML(f"result table {x}"))
    print(t)

```

I was surprised that the toc2 would not pick up on them.

Checking the code, I see that the header-lookup is limited to [$('.text_cell_render')](https://github.com/ipython-contrib/jupyter_contrib_nbextensions/blob/4906812149dcaa66250304ba541c24612dc74fa6/src/jupyter_contrib_nbextensions/nbextensions/toc2/toc2.js#L560). Is there a specific reason for that? Or could this extended easily to `output_subarea` elements ? 

I tried to play around a bit, but I don't understand the piece of code selecting the headers (`.find('[id]:header:not(:has(.tocSkip))');`)

**Comment by gwerbin** [comment]
I think this is by design. My understanding of the notebook workflow is that you use a Markdown input cell to put a heading in the notebook, then you have the code generating that heading's plot underneath it. I can see why that's limiting for your case.

**Comment by mpschr** [comment]
Yes, I used markdown in first place but thereafter assumed that all headers tags in a notebook would be captured. 

I could work out a solution for the report I was working on: for any one interested, it's in the referenced pull request, above. I imagine it could be of wider interest to incorporate the feature.
